drake/util/TrigPoly.h

```diff
@@ -3,8 +3,30 @@
 
 #include "drake/util/Polynomial.h"
 #include <map>
-//#include <iostream>  // just for debugging
 
+/// A scalar multi-variate polynomial containing sines and cosines.
+/**
+ * TrigPoly represents a Polynomial some of whose variables actually represent
+ * the sines or cosines of other variables.  Sines and cosines of first-order
+ * polynomials (affine combinations of variables) are decomposed into
+ * polynomials of the sines and cosines of individual variables via the
+ * Prosthaphaeresis formulae.
```

Comment:
Yes.  I realized after I logged off that it is not enforced programmatically.  It’s just that we need for this to be true in order to do any analysis on it.  

In matlab, we have a method which performs the runtime check:
https://github.com/RobotLocomotion/drake/blob/master/drake/util/TrigPoly.m#L85

(sorry for the confusion)

- Russ